## Задание 3 (Дедубликация )
### Описание задания:
Вам нужно написать код на python, который сделает дедубликацию торговых точек. Вы можете использовать свой алгоритм и/или сторонние(внешние) сервисы.
Пожалуйста, предоставьте ссылку на репозиторий Github(Bitbucket), где будет находиться Ваш код и конечный результат(SQL таблицы с дедублицированными данными).

SQL таблицы - https://drive.google.com/file/d/0B6nelhokQOrRZkMzWHgzSWFBaUE/view

Вам нужно будет заполнить таблицу “outlets_clean” и обновить внешний ключ в таблице “outlets”(колонка outlet_clean_id). Для точек в таблице “outlets”, которые Вы не можете дедублицировать, проставьте в колонке outlet_clean_id значение NULL.


Выполнил Гусев Иван.

### Предварительная подготовка:
В фале находился скрипт для создания БД. Запустил скрипт и выгрузил в csv файл outlets_h.csv

In [1]:
import pandas as pd
import re
import numpy as np

In [2]:
data = pd.read_csv('outlets_h.csv', sep=';', encoding ='mbcs')
data.head(50)

id Город дистрибьютора                    Торг_точка_грязная  \
0    1      Ростов-на-Дону                      ИП Гаврилов С.А.   
1    2      Ростов-на-Дону                           ИП Малышева   
2    3      Ростов-на-Дону                        Абасян А.П. ИП   
3    4      Ростов-на-Дону             Азатян Н,С. ИП ул. Ленина   
4    5      Ростов-на-Дону                          Айдинов В.Я.   
5    6      Ростов-на-Дону                      Алексеева Л П ИП   
6    7      Ростов-на-Дону                      Алексеева Л П ИП   
7    8      Ростов-на-Дону                        ИП Алиева Т.М.   
8    9      Ростов-на-Дону                       Алпатова В В ИП   
9   10      Ростов-на-Дону                       Алпатова В В ИП   
10  11      Ростов-на-Дону                      Алпатова В.В. ИП   
11  12      Ростов-на-Дону           Алпатова В.В. ИП "Ассорти+"   
12  13      Ростов-на-Дону  Алпатова В.В. ИП ул.Гвардейская 34-а   
13  14      Ростов-на-Дону                    ООО "Альтернатива"   
14  15      Ростов-на-Дону              Альянс ООО г.Новочеркаск   
15  16      Ростов-на-Дону                     Андреасян А.А, ИП   
16  17      Ростов-на-Дону                         Анохин С М ИП   
17  18      Ростов-на-Дону          Антипова И.Д. п.Васильевский   
18  19      Ростов-на-Дону                              Арег ООО   
19  20      Ростов-на-Дону                              Армада-Н   
20  21      Ростов-на-Дону                             Арнес ООО   
21  22      Ростов-на-Дону                          ИП Арустамян   
22  23      Ростов-на-Дону     Асмаева Т.А. ИП Маг. Овощи-фрукты   
23  24      Ростов-на-Дону                            Асриян Б Х   
24  25      Ростов-на-Дону                          ООО "Ассоль"   
25  26      Ростов-на-Дону                        Атаари К.Г. ИП   
26  27      Ростов-на-Дону            Бабкова О.Е. ИП х.Грушевка   
27  28      Ростов-на-Дону                                  Бага   
28  29      Ростов-на-Дону                           Баулин Н.Е.   
29  30      Ростов-на-Дону            Баусова И.В.ИП м-н "Ирина"   
30  31      Ростов-на-Дону                         ИП Безбородов   
31  32      Ростов-на-Дону                      Беленков И.А. ИП   
32  33      Ростов-на-Дону                             Беник пок   
33  34      Ростов-на-Дону                            Беник пок.   
34  35      Ростов-на-Дону                            Беник пок.   
35  36      Ростов-на-Дону      Бирюкова Л.Н.ИП Фурманова, тп 20   
36  37      Ростов-на-Дону                       Блок ЗАО м-н №7   
37  38      Ростов-на-Дону                     Болдырева И.Н. ИП   
38  39      Ростов-на-Дону                            ИП Бородин   
39  40      Ростов-на-Дону                           Боря Дубров   
40  41      Ростов-на-Дону                           Боря Дубров   
41  42      Ростов-на-Дону                           Боря Дубров   
42  43      Ростов-на-Дону                           Боря липтон   
43  44      Ростов-на-Дону                                  Брак   
44  45      Ростов-на-Дону                     Брижанова Г.И. ИП   
45  46      Ростов-на-Дону                     ИП Бросалина Л.Н.   
46  47      Ростов-на-Дону                          Бунич Г.Ф.ИП   
47  48      Ростов-на-Дону                      Буркутов С.И. ИП   
48  49      Ростов-на-Дону           Бычихин В.И. ИП ул.Новая 26   
49  50      Ростов-на-Дону                             Валео ООО   

                            Торг_точка_грязная_адрес  outlet_clean_id  
0                           п.Глубокий, ул.Артёма 44              NaN  
1                                                  -              NaN  
2                                              он же              NaN  
3                                              он же              NaN  
4                                              он же              NaN  
5                                      Буденного 161              NaN  
6                              

In [3]:
data['name_nwe'] = data['Торг_точка_грязная']
data['adr_nwe'] = data['Торг_точка_грязная_адрес']

In [4]:
data_re = data

In [5]:
# Избавимся в столбцах с адресом и названия точек от спец символов уберем лишние пробелы и приведем все к нижнему регистру:
data_re['name_nwe'] = data_re['name_nwe'].apply(lambda x: re.sub('\W+',' ', x ) )
data_re['adr_nwe'] = data_re['adr_nwe'].apply(lambda x: re.sub('\W+',' ', x ) )
data_re['name_nwe'] = data_re['name_nwe'].str.lower()
data_re['adr_nwe'] = data_re['adr_nwe'].str.lower()
data['name_nwe'] = data['name_nwe'].str.strip()
data['adr_nwe'] = data['adr_nwe'].str.strip()

In [6]:
data_re.head(45)

id Город дистрибьютора                    Торг_точка_грязная  \
0    1      Ростов-на-Дону                      ИП Гаврилов С.А.   
1    2      Ростов-на-Дону                           ИП Малышева   
2    3      Ростов-на-Дону                        Абасян А.П. ИП   
3    4      Ростов-на-Дону             Азатян Н,С. ИП ул. Ленина   
4    5      Ростов-на-Дону                          Айдинов В.Я.   
5    6      Ростов-на-Дону                      Алексеева Л П ИП   
6    7      Ростов-на-Дону                      Алексеева Л П ИП   
7    8      Ростов-на-Дону                        ИП Алиева Т.М.   
8    9      Ростов-на-Дону                       Алпатова В В ИП   
9   10      Ростов-на-Дону                       Алпатова В В ИП   
10  11      Ростов-на-Дону                      Алпатова В.В. ИП   
11  12      Ростов-на-Дону           Алпатова В.В. ИП "Ассорти+"   
12  13      Ростов-на-Дону  Алпатова В.В. ИП ул.Гвардейская 34-а   
13  14      Ростов-на-Дону                    ООО "Альтернатива"   
14  15      Ростов-на-Дону              Альянс ООО г.Новочеркаск   
15  16      Ростов-на-Дону                     Андреасян А.А, ИП   
16  17      Ростов-на-Дону                         Анохин С М ИП   
17  18      Ростов-на-Дону          Антипова И.Д. п.Васильевский   
18  19      Ростов-на-Дону                              Арег ООО   
19  20      Ростов-на-Дону                              Армада-Н   
20  21      Ростов-на-Дону                             Арнес ООО   
21  22      Ростов-на-Дону                          ИП Арустамян   
22  23      Ростов-на-Дону     Асмаева Т.А. ИП Маг. Овощи-фрукты   
23  24      Ростов-на-Дону                            Асриян Б Х   
24  25      Ростов-на-Дону                          ООО "Ассоль"   
25  26      Ростов-на-Дону                        Атаари К.Г. ИП   
26  27      Ростов-на-Дону            Бабкова О.Е. ИП х.Грушевка   
27  28      Ростов-на-Дону                                  Бага   
28  29      Ростов-на-Дону                           Баулин Н.Е.   
29  30      Ростов-на-Дону            Баусова И.В.ИП м-н "Ирина"   
30  31      Ростов-на-Дону                         ИП Безбородов   
31  32      Ростов-на-Дону                      Беленков И.А. ИП   
32  33      Ростов-на-Дону                             Беник пок   
33  34      Ростов-на-Дону                            Беник пок.   
34  35      Ростов-на-Дону                            Беник пок.   
35  36      Ростов-на-Дону      Бирюкова Л.Н.ИП Фурманова, тп 20   
36  37      Ростов-на-Дону                       Блок ЗАО м-н №7   
37  38      Ростов-на-Дону                     Болдырева И.Н. ИП   
38  39      Ростов-на-Дону                            ИП Бородин   
39  40      Ростов-на-Дону                           Боря Дубров   
40  41      Ростов-на-Дону                           Боря Дубров   
41  42      Ростов-на-Дону                           Боря Дубров   
42  43      Ростов-на-Дону                           Боря липтон   
43  44      Ростов-на-Дону                                  Брак   
44  45      Ростов-на-Дону                     Брижанова Г.И. ИП   

                            Торг_точка_грязная_адрес  outlet_clean_id  \
0                           п.Глубокий, ул.Артёма 44              NaN   
1                                                  -              NaN   
2                                              он же              NaN   
3                                              он же              NaN   
4                                              он же              NaN   
5                                      Буденного 161              NaN   
6                                              он же              NaN   
7                       Ип Алиева Т.М. п.Газетный 39              NaN   
8                                              он же              NaN   
9                                ул Гвардейская 34-а              NaN   
10                                             он же              NaN   

In [7]:
data_re[['name_nwe','adr_nwe']].duplicated().sum()


5147

Уже имеем 5147 явных дубликатов.

Было бы хорошо уточнить у составителя таблицы как она заполнялась, в какой последовательности. Возможно дало бы нам более ясное понимание что означает "-" в строках с адресом и по какому принципу заполнялось поле адреса с "он же".

За неимением такой информации предположим что "-" это если адрес совсем отсутствует, а "он же" актуален для точек расположенных выше в файле.

In [8]:
for i in range(len(data_re)):
    if data_re['adr_nwe'][i] != 'он же':
        z = data_re['adr_nwe'][i]
    else:
       data_re['adr_nwe'][i] = z 

C:\Users\esluc\AppData\Local\Temp\ipykernel_11740\2597019675.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_re['adr_nwe'][i] = z


In [9]:
data_re.head(45)

id Город дистрибьютора                    Торг_точка_грязная  \
0    1      Ростов-на-Дону                      ИП Гаврилов С.А.   
1    2      Ростов-на-Дону                           ИП Малышева   
2    3      Ростов-на-Дону                        Абасян А.П. ИП   
3    4      Ростов-на-Дону             Азатян Н,С. ИП ул. Ленина   
4    5      Ростов-на-Дону                          Айдинов В.Я.   
5    6      Ростов-на-Дону                      Алексеева Л П ИП   
6    7      Ростов-на-Дону                      Алексеева Л П ИП   
7    8      Ростов-на-Дону                        ИП Алиева Т.М.   
8    9      Ростов-на-Дону                       Алпатова В В ИП   
9   10      Ростов-на-Дону                       Алпатова В В ИП   
10  11      Ростов-на-Дону                      Алпатова В.В. ИП   
11  12      Ростов-на-Дону           Алпатова В.В. ИП "Ассорти+"   
12  13      Ростов-на-Дону  Алпатова В.В. ИП ул.Гвардейская 34-а   
13  14      Ростов-на-Дону                    ООО "Альтернатива"   
14  15      Ростов-на-Дону              Альянс ООО г.Новочеркаск   
15  16      Ростов-на-Дону                     Андреасян А.А, ИП   
16  17      Ростов-на-Дону                         Анохин С М ИП   
17  18      Ростов-на-Дону          Антипова И.Д. п.Васильевский   
18  19      Ростов-на-Дону                              Арег ООО   
19  20      Ростов-на-Дону                              Армада-Н   
20  21      Ростов-на-Дону                             Арнес ООО   
21  22      Ростов-на-Дону                          ИП Арустамян   
22  23      Ростов-на-Дону     Асмаева Т.А. ИП Маг. Овощи-фрукты   
23  24      Ростов-на-Дону                            Асриян Б Х   
24  25      Ростов-на-Дону                          ООО "Ассоль"   
25  26      Ростов-на-Дону                        Атаари К.Г. ИП   
26  27      Ростов-на-Дону            Бабкова О.Е. ИП х.Грушевка   
27  28      Ростов-на-Дону                                  Бага   
28  29      Ростов-на-Дону                           Баулин Н.Е.   
29  30      Ростов-на-Дону            Баусова И.В.ИП м-н "Ирина"   
30  31      Ростов-на-Дону                         ИП Безбородов   
31  32      Ростов-на-Дону                      Беленков И.А. ИП   
32  33      Ростов-на-Дону                             Беник пок   
33  34      Ростов-на-Дону                            Беник пок.   
34  35      Ростов-на-Дону                            Беник пок.   
35  36      Ростов-на-Дону      Бирюкова Л.Н.ИП Фурманова, тп 20   
36  37      Ростов-на-Дону                       Блок ЗАО м-н №7   
37  38      Ростов-на-Дону                     Болдырева И.Н. ИП   
38  39      Ростов-на-Дону                            ИП Бородин   
39  40      Ростов-на-Дону                           Боря Дубров   
40  41      Ростов-на-Дону                           Боря Дубров   
41  42      Ростов-на-Дону                           Боря Дубров   
42  43      Ростов-на-Дону                           Боря липтон   
43  44      Ростов-на-Дону                                  Брак   
44  45      Ростов-на-Дону                     Брижанова Г.И. ИП   

                            Торг_точка_грязная_адрес  outlet_clean_id  \
0                           п.Глубокий, ул.Артёма 44              NaN   
1                                                  -              NaN   
2                                              он же              NaN   
3                                              он же              NaN   
4                                              он же              NaN   
5                                      Буденного 161              NaN   
6                                              он же              NaN   
7                       Ип Алиева Т.М. п.Газетный 39              NaN   
8                                              он же              NaN   
9                                ул Гвардейская 34-а              NaN   
10                                             он же              NaN   

In [10]:
data_re[['name_nwe','adr_nwe']].duplicated().sum()

5228

In [11]:
print(data_re['name_nwe'].duplicated().sum())


10226


Можем наблюдать что дубликатов названий гораздо больше чем названий в связке с адресами, при получении информации о заполнении таблицы можно будет увеличить эффективность практически в 2 раза. Без нее мы можем не корректно заполнить адреса для некоторых точек. В данном случае считаю что отсутсвсие информации лучше чем не корректная.

In [12]:
data_re['dub_y'] = data_re[['name_nwe','adr_nwe']].duplicated() 

In [13]:
# в датафреме ниже мы получили явные дубликаты, их гораздо меньше чем на ссамом деле но это практически без потери либо
# с отстутсвием не корректной информации
data_re.head(45)

id Город дистрибьютора                    Торг_точка_грязная  \
0    1      Ростов-на-Дону                      ИП Гаврилов С.А.   
1    2      Ростов-на-Дону                           ИП Малышева   
2    3      Ростов-на-Дону                        Абасян А.П. ИП   
3    4      Ростов-на-Дону             Азатян Н,С. ИП ул. Ленина   
4    5      Ростов-на-Дону                          Айдинов В.Я.   
5    6      Ростов-на-Дону                      Алексеева Л П ИП   
6    7      Ростов-на-Дону                      Алексеева Л П ИП   
7    8      Ростов-на-Дону                        ИП Алиева Т.М.   
8    9      Ростов-на-Дону                       Алпатова В В ИП   
9   10      Ростов-на-Дону                       Алпатова В В ИП   
10  11      Ростов-на-Дону                      Алпатова В.В. ИП   
11  12      Ростов-на-Дону           Алпатова В.В. ИП "Ассорти+"   
12  13      Ростов-на-Дону  Алпатова В.В. ИП ул.Гвардейская 34-а   
13  14      Ростов-на-Дону                    ООО "Альтернатива"   
14  15      Ростов-на-Дону              Альянс ООО г.Новочеркаск   
15  16      Ростов-на-Дону                     Андреасян А.А, ИП   
16  17      Ростов-на-Дону                         Анохин С М ИП   
17  18      Ростов-на-Дону          Антипова И.Д. п.Васильевский   
18  19      Ростов-на-Дону                              Арег ООО   
19  20      Ростов-на-Дону                              Армада-Н   
20  21      Ростов-на-Дону                             Арнес ООО   
21  22      Ростов-на-Дону                          ИП Арустамян   
22  23      Ростов-на-Дону     Асмаева Т.А. ИП Маг. Овощи-фрукты   
23  24      Ростов-на-Дону                            Асриян Б Х   
24  25      Ростов-на-Дону                          ООО "Ассоль"   
25  26      Ростов-на-Дону                        Атаари К.Г. ИП   
26  27      Ростов-на-Дону            Бабкова О.Е. ИП х.Грушевка   
27  28      Ростов-на-Дону                                  Бага   
28  29      Ростов-на-Дону                           Баулин Н.Е.   
29  30      Ростов-на-Дону            Баусова И.В.ИП м-н "Ирина"   
30  31      Ростов-на-Дону                         ИП Безбородов   
31  32      Ростов-на-Дону                      Беленков И.А. ИП   
32  33      Ростов-на-Дону                             Беник пок   
33  34      Ростов-на-Дону                            Беник пок.   
34  35      Ростов-на-Дону                            Беник пок.   
35  36      Ростов-на-Дону      Бирюкова Л.Н.ИП Фурманова, тп 20   
36  37      Ростов-на-Дону                       Блок ЗАО м-н №7   
37  38      Ростов-на-Дону                     Болдырева И.Н. ИП   
38  39      Ростов-на-Дону                            ИП Бородин   
39  40      Ростов-на-Дону                           Боря Дубров   
40  41      Ростов-на-Дону                           Боря Дубров   
41  42      Ростов-на-Дону                           Боря Дубров   
42  43      Ростов-на-Дону                           Боря липтон   
43  44      Ростов-на-Дону                                  Брак   
44  45      Ростов-на-Дону                     Брижанова Г.И. ИП   

                            Торг_точка_грязная_адрес  outlet_clean_id  \
0                           п.Глубокий, ул.Артёма 44              NaN   
1                                                  -              NaN   
2                                              он же              NaN   
3                                              он же              NaN   
4                                              он же              NaN   
5                                      Буденного 161              NaN   
6                                              он же              NaN   
7                       Ип Алиева Т.М. п.Газетный 39              NaN   
8                                              он же              NaN   
9                                ул Гвардейская 34-а              NaN   
10                                             он же              NaN   

Если потеря и искажение данных не так критично то можем проделать следующее:

Предполагая что если в столбце адреса стоит "он же" - это дубликат предыдушей записи не смотря на название органаизации.
Так же дубликатами будем считать одинаковые названия ИП и одинаковые адреса.



In [14]:
data_re['dub_name'] = data['name_nwe'].duplicated()
data_re['dub_adr'] = data['adr_nwe'].duplicated()
data_re['dub_same'] = data['Торг_точка_грязная_адрес'] == 'он же'

In [15]:
data_re.head(10)

id Город дистрибьютора         Торг_точка_грязная  \
0   1      Ростов-на-Дону           ИП Гаврилов С.А.   
1   2      Ростов-на-Дону                ИП Малышева   
2   3      Ростов-на-Дону             Абасян А.П. ИП   
3   4      Ростов-на-Дону  Азатян Н,С. ИП ул. Ленина   
4   5      Ростов-на-Дону               Айдинов В.Я.   
5   6      Ростов-на-Дону           Алексеева Л П ИП   
6   7      Ростов-на-Дону           Алексеева Л П ИП   
7   8      Ростов-на-Дону             ИП Алиева Т.М.   
8   9      Ростов-на-Дону            Алпатова В В ИП   
9  10      Ростов-на-Дону            Алпатова В В ИП   

       Торг_точка_грязная_адрес  outlet_clean_id                 name_nwe  \
0      п.Глубокий, ул.Артёма 44              NaN          ип гаврилов с а   
1                             -              NaN              ип малышева   
2                         он же              NaN            абасян а п ип   
3                         он же              NaN  азатян н с ип ул ленина   
4                         он же              NaN              айдинов в я   
5                 Буденного 161              NaN         алексеева л п ип   
6                         он же              NaN         алексеева л п ип   
7  Ип Алиева Т.М. п.Газетный 39              NaN            ип алиева т м   
8                         он же              NaN          алпатова в в ип   
9           ул Гвардейская 34-а              NaN          алпатова в в ип   

                       adr_nwe  dub_y  dub_name  dub_adr  dub_same  
0      п глубокий ул артёма 44  False     False    False     False  
1                               False     False    False     False  
2                               False     False     True      True  
3                               False     False     True      True  
4                               False     False     True      True  
5                буденного 161  False     False    False     False  
6                буденного 161   True      True     True      True  
7  ип алиева т м п газетный 39  False     False    False     False  
8  ип алиева т м п газетный 39  False     False     True      True  
9          ул гвардейская 34 а  False      True    False     False

Далее будем считать если хотя бы в одном столбце есть метка что строка является дубликатом будем считать что это дубликат

Либо если изменятся входные данные можем использовать разные сочетания данных столбцов для определения дубликатов и изменить результат.

In [16]:
data_re['dub_total'] = data_re['dub_y']+data_re['dub_name']+data_re['dub_adr']+data_re['dub_same']

D:\pyton\Anacocda\envs\nwe_praktikum_env_win_new\lib\site-packages\pandas\core\computation\expressions.py:204: UserWarning: evaluating in Python space because the '+' operator is not supported by numexpr for the bool dtype, use '|' instead
  warnings.warn(


In [52]:
data_re.head()

id Город дистрибьютора         Торг_точка_грязная  \
0   1      Ростов-на-Дону           ИП Гаврилов С.А.   
1   2      Ростов-на-Дону                ИП Малышева   
2   3      Ростов-на-Дону             Абасян А.П. ИП   
3   4      Ростов-на-Дону  Азатян Н,С. ИП ул. Ленина   
4   5      Ростов-на-Дону               Айдинов В.Я.   

   Торг_точка_грязная_адрес  outlet_clean_id                 name_nwe  \
0  п.Глубокий, ул.Артёма 44              0.0          ип гаврилов с а   
1                         -              1.0              ип малышева   
2                     он же              NaN            абасян а п ип   
3                     он же              NaN  азатян н с ип ул ленина   
4                     он же              NaN              айдинов в я   

                   adr_nwe  dub_y  dub_name  dub_adr  dub_same  dub_total  
0  п глубокий ул артёма 44  False     False    False     False      False  
1                           False     False    False     False      False  
2                           False     False     True      True       True  
3                           False     False     True      True       True  
4                           False     False     True      True       True

In [18]:
len(data_re) - data_re['dub_total'].sum()

2848

Получаем всего 2848 уникальных значений.

In [19]:
data_re['outlet_clean_id'] = range(len(data_re))
idx = data_re.query("dub_total == True").index
data_re.loc[idx, 'outlet_clean_id'] = np.NaN


In [20]:
data_re.columns

Index(['id', 'Город дистрибьютора', 'Торг_точка_грязная',
       'Торг_точка_грязная_адрес', 'outlet_clean_id', 'name_nwe', 'adr_nwe',
       'dub_y', 'dub_name', 'dub_adr', 'dub_same', 'dub_total'],
      dtype='object')

In [21]:

outlets = data_re[['id', 'Город дистрибьютора', 'Торг_точка_грязная',
       'Торг_точка_грязная_адрес', 'outlet_clean_id', 'name_nwe', 'adr_nwe']]
outlets.to_csv('outlets', encoding='utf-8' )
outlets.head(10)

id Город дистрибьютора         Торг_точка_грязная  \
0   1      Ростов-на-Дону           ИП Гаврилов С.А.   
1   2      Ростов-на-Дону                ИП Малышева   
2   3      Ростов-на-Дону             Абасян А.П. ИП   
3   4      Ростов-на-Дону  Азатян Н,С. ИП ул. Ленина   
4   5      Ростов-на-Дону               Айдинов В.Я.   
5   6      Ростов-на-Дону           Алексеева Л П ИП   
6   7      Ростов-на-Дону           Алексеева Л П ИП   
7   8      Ростов-на-Дону             ИП Алиева Т.М.   
8   9      Ростов-на-Дону            Алпатова В В ИП   
9  10      Ростов-на-Дону            Алпатова В В ИП   

       Торг_точка_грязная_адрес  outlet_clean_id                 name_nwe  \
0      п.Глубокий, ул.Артёма 44              0.0          ип гаврилов с а   
1                             -              1.0              ип малышева   
2                         он же              NaN            абасян а п ип   
3                         он же              NaN  азатян н с ип ул ленина   
4                         он же              NaN              айдинов в я   
5                 Буденного 161              5.0         алексеева л п ип   
6                         он же              NaN         алексеева л п ип   
7  Ип Алиева Т.М. п.Газетный 39              7.0            ип алиева т м   
8                         он же              NaN          алпатова в в ип   
9           ул Гвардейская 34-а              NaN          алпатова в в ип   

                       adr_nwe  
0      п глубокий ул артёма 44  
1                               
2                               
3                               
4                               
5                буденного 161  
6                буденного 161  
7  ип алиева т м п газетный 39  
8  ип алиева т м п газетный 39  
9          ул гвардейская 34 а

In [24]:
outlets_clean = data_re[~data_re['outlet_clean_id'].isna()][['adr_nwe','outlet_clean_id']]
outlets_clean.rename(columns={'adr_nwe':'Торг_точка_чистый_адрес','outlet_clean_id':'id'})
outlets_clean.to_csv('outlets_clean', encoding='utf-8' )

Импортировать таблицу можно с помощью следующего кода:

In [23]:
# Загрузка таблиц в MySQL
import sqlalchemy
database_username = 'ENTER USERNAME'
database_password = 'ENTER USERNAME PASSWORD'
database_ip       = 'ENTER DATABASE IP'
database_name     = 'ENTER DATABASE NAME'
database_connection = sqlalchemy.create_engine('mysql+mysqlconnector://{0}:{1}@{2}/{3}'.
                                               format(database_username, database_password, 
                                                      database_ip, database_name))
outlets.to_sql(con=database_connection, name='outlets_clean', if_exists='replace')
outlets_clean.to_sql(con=database_connection, name='outlets_clean', if_exists='replace')

ModuleNotFoundError: No module named 'mysql'

Затем отправить запрос либо сделать в базе.
```
ALTER TABLE outlets_clean ADD id SMALLINT UNSIGNED NOT NULL DEFAULT 0;
ALTER TABLE outlets_clean ADD CONSTRAINT fk_id FOREIGN KEY (id) REFERENCES outlets(outlet_clean_id);
``